src/interfaces/token/IHook.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.5.0;
+
+import {IERC165} from "src/interfaces/IERC7575.sol";
+
+struct HookData {
+    bytes16 from;
+    bytes16 to;
+}
+
+uint8 constant SUCCESS_CODE = 0;
+string constant SUCCESS_MESSAGE = "transfer-allowed";
+
+interface IHook is IERC165 {
```

Comment:
I was thinking that `Hook` is a very generic term. Here we define an interface of a hook that has very specific  functions.
What do you think about defining it like [that](https://docs.stackup.sh/docs/erc-7579#hooks) 

Then we can have  multiple multiple implementations such as `AuthTransferHook` , `TransferHook` , `RestrictionsHook` ( which is the manager itself  ). 

Also in the `TrancheToken` you won't have a single `hook`  but rather a mapping of hooks  mapped by name and address and you can call different hook within the `TrancheToken` based on their name and just call `preCheck` , `postCheck`.

This is a very rough idea.